src/core_modules/capture-core/components/WidgetEnrollment/hooks/useEnrollment.js

```diff
@@ -41,6 +43,12 @@ export const useEnrollment = ({
         }
     }, [setEnrollment, data]);
 
+    useEffect(() => {
+        if (externalData?.status?.value) {
+            setEnrollment(e => ({ ...e, status: externalData?.status?.value }));
+        }
+    }, [setEnrollment, externalData?.status?.value]);
```

Comment:
```suggestion
    }, [setEnrollment, externalData?.status]);
```
To make sure this useEffect is always triggered when we set the status we should use the object reference instead of the actual value